safe_init/secrets.py

```diff
@@ -35,6 +45,44 @@ def resolve_secrets() -> Mapping[str, str | None]:
         The resolved secrets as a dictionary.
     """
     common_secret_arn_prefix = os.getenv("SAFE_INIT_SECRET_ARN_PREFIX")
+    secret_arns = gather_secret_arns(common_secret_arn_prefix)
+
+    # Try to get secret values from Redis cache and identify secrets that are not in cache
+    secrets, secrets_not_in_cache = get_secrets_from_cache(secret_arns)
+
+    if secrets_not_in_cache:
+        try:
+            fetched_secrets, errors = get_secrets_from_secrets_manager(secrets_not_in_cache)
+
+            if errors:
+                error_message = f"Failed to retrieve secrets: {errors}"
+                raise SecretResolutionError(error_message, errors)  # noqa: TRY301
+
+            for secret_arn, secret_value in fetched_secrets.items():
+                save_secret_in_cache(secret_arn, secret_value)
+                secrets[secret_arn] = secret_value
+        except Exception as e:
+            if bool_env("SAFE_INIT_FAIL_ON_SECRET_RESOLUTION_ERROR"):
+                raise
+            log_warning("Failed to resolve some secrets", exc_info=e)
```

Comment:
Should this be log_error?

Reply:
It's consistent with the current implementation, which is a warning. If we want it to fail, there's the env var that raises if secret resolution fails